### PR TITLE
Add additional named parameters and job parameters support to DatabricksRunNowOperator

### PR DESCRIPTION
**Pull request by pranshupand-db**
- Changes
  -  Added additional named parameters support to `DatabricksRunNowOperator`: 
      - `dbt_commands`
      - `job_parameters` 
 - How is it tested
   - Updated the tests to check the new changes 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/databricks/operators/databricks.py [hunk 1]
```diff
@@ -731,6 +740,17 @@ class DatabricksRunNowOperator(BaseOperator):
         It must exist only one job with the specified name.
         ``job_id`` and ``job_name`` are mutually exclusive.
         This field will be templated.
+
+    :param job_parameters: A dict from keys to values that override or augment the job's
```
Comment (gaborratky-db) [inline providers/src/airflow/providers/databricks/operators/databricks.py:740]: Should we also add similar docs for `dbt_commands`?

File: providers/src/airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -711,12 +717,15 @@ class DatabricksRunNowOperator(BaseOperator):
...
         - ``job_name``
+        - ``job_parameters``
         - ``json``
+        - ``dbt_commands``
         - ``notebook_params``
         - ``python_params``
         - ``python_named_parameters``
         - ``jar_params``
         - ``spark_submit_params``
+        - ``sql_params``
```
Comment (gaborratky-db) [inline providers/src/airflow/providers/databricks/operators/databricks.py]: We shouldn't add support for `sql_params` at this point. They are [deprecated in the API](https://docs.databricks.com/api/workspace/jobs/runnow#sql_params), and passing `job_parameters` to the job run will automatically push them to every SQL task in the job.

**Review by gaborratky-db — CHANGES_REQUESTED** [review]
Thanks for contributing this, @pranshupand-db! A few requests before the core team reviews the PR:
 * We should **not** add support for `sql_params`. See my inline comment for more details.
 * We should add support for `dbt_commands` because even though it is being deprecated along with `sql_params`, using `job_parameters` for dbt tasks is more complicated because you have to use dynamic value references so it's not a drop-in replacement.
 * Let's make sure documentation is present for `dbt_commands`, similar to `job_parameters`.

File: providers/src/airflow/providers/databricks/operators/databricks.py (hunk 2 above)
Comment (pranshupand-db) [reply providers/src/airflow/providers/databricks/operators/databricks.py]: Sure.

File: providers/src/airflow/providers/databricks/operators/databricks.py [hunk 3]
```diff
@@ -741,6 +758,14 @@ class DatabricksRunNowOperator(BaseOperator):
         .. seealso::
             For more information about templating see :ref:`concepts:jinja-templating`.
             https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsRunNow
+
+    :param dbt_commands: A list containing the dbt commands to run using dbt command line
+        interface(CLI). This field will be templated.
```
Comment (gaborratky-db) [inline providers/src/airflow/providers/databricks/operators/databricks.py]:
```suggestion
    :param dbt_commands: A list containing the dbt commands to run using the dbt command line
        interface (CLI). This field will be templated.
```

**Review by gaborratky-db — APPROVED** [review]
LGTM with small nits

**Review by alexott — APPROVED** [review]

**Comment by pranshupand-db** [comment]
Please review @potiuk

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Can you please rebase @pranshupand-db -> for some reason the "ci" workflow does not start with this one, and it's ~ 50 commits behind.

**Comment by pranshupand-db** [comment]
@potiuk done rebasing.

**Comment by potiuk** [comment]
There are some errors in generated docs from the docstring

**Comment by pranshupand-db** [comment]
@potiuk Please merge

**Comment by gaborratky-db** [comment]
@potiuk, please let us know if you need any other changes here or if this is ready to be merged. Thanks!

**Comment by potiuk** [comment]
@gaborratky-db -> I could not merge it (checking for mergability for eer), looks like some strange state of the PR - I tried to close/reopen it but then I cannot reopen it as usual. So I think you need to push your changes again and reopen it.

**Comment by potiuk** [comment]
hello

**Pull request closed without merging**